packages/jest-snapshot/src/State.ts

```diff
@@ -27,6 +27,7 @@ export type SnapshotStateOptions = {
   prettierPath: string;
   expand?: boolean;
   snapshotFormat: PrettyFormatOptions;
+  rootDir: string;
```

Comment:
it's optional as you might have a different version of `jest-snapshot` (or `jest-snapshot` is used in a different context than circus/jasmine (e.g. https://github.com/nicolo-ribaudo/jest-light-runner/blob/220ea9556a62fbf26bf76e02ba241ee68944f1c4/src/worker-runner.js#L72-L76)).

If missing, I guess we should fall back to `dirname` still?

Reply:
I don't think the rootDir approach is a good mid-term solution. It wouldn't work for the case with inline config like in the test and it also wouldn't work for the React repo since they have  a separate `preprocessor.js` file that declares the plugins inline i.e.facebook/react doesn't have a babel.config.js either.

Maybe just include the JSX plugin in the current plugin list (unless we have a `.ts` file) and call it a day?

Reply:
Yeah, possibly. What about flow?

Should _also_ do this, tho

Reply:
Maybe accept an optional `ast` (that matches babels AST) in [`TransformedSource`](https://jestjs.io/docs/next/code-transformation) and use that directly if we have it? 

That way we don't have to guess plugins and also avoid an additional parser step. People using `babel.transform`  wouldn't have to change anything since `ast` is already included